data/src/main/java/com/github/leuludyha/data/api/Raw.kt

```diff
@@ -1,5 +1,9 @@
 package com.github.leuludyha.data.api
 
+/**
+ * Common interface for raw API responses types (used internally, that needs to be
+ * converted before being passed to the model)
+ */
 interface Raw<T> {
-    fun toModel(libraryApi: LibraryApi): T
+    fun toModel(libraryApi: LibraryApi): T?
```

Comment:
If the key is not valid (for example doesn't start with "/works/" prefix for a work). In principle it shouldn't happen but we never know.

Reply:
But maybe we should catch this error if it occurs and let it crash instead of carrying a null object which we may not find a source for it. Null objects carry little information is what I'm trying to say

Reply:
My bad I didn't think about it but in any case what is transmitted to the model is a `Result` (see `rawResponseToModelResult` in `ApiHelper.kt`).

`Result.Success(data)` if no error occurs, `Result.Error(error)` if an error occurs. So the final api exposed to the model doesn't contain any null values. However I agree I could already return `Result<T>` in `toModel` but as it is used only internally by the API it's ok like that, we will never deal with that at the app or model layers.